Sources/TABResourceLoader/Extensions/Dictionary+Additions.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  Dictionary+Additions.swift
+//  TABResourceLoader
+//
+//  Created by John Sanderson on 30/01/2018.
+//  Copyright © 2018 Luciano Marisi. All rights reserved.
+//
+
+import Foundation
+
+public extension Dictionary where Key == String, Value == CustomStringConvertible {
+
+  /// Returns dictionary as string used to POST Form Data
+  public var formDataPostString: String {
+    var data = [String]()
+    forEach { (key, value) in
+      let stringValue = String(describing: value)
+      data.append(key + "=" + stringValue)
+    }
+    return data.map(String.init(_:)).joined(separator: "&")
```

Comment:
Why is the `map(String.init(_:))` needed? Could the whole thing be done with a single `reduce` call on the dictionary?

Reply:
yeah a single reduce would work better. but anyway `data` is of type `[String]` so `data.map(String.init)` will just return the same as `data`. won't it?

Reply:
Updated to use reduce